napari/utils/events/dataclass.py

```diff
@@ -181,12 +181,15 @@ def set_with_events(self: C, name: str, value: Any) -> None:
     # if different we emit the event with new value
     after = getattr(self, name)
     if isinstance(before, np.ndarray) or isinstance(after, np.ndarray):
-        different = np.any(before != after)
```

Comment:
I found that this comparison could return false positive. for example when before is `1` and after is `np.array([1,1])`

Reply:
That's one reason why I suggested using no.array_equal in a previous comment, but also, because this use of np.any and piecewise comparison will throw a warning

Reply:
https://github.com/napari/napari/pull/1966#discussion_r532115845

Reply:
Ok